### PR TITLE
Removes JSI overrides from ReactCommon

### LINKED ISSUE #9703: Add PropNameID::fromSymbol to JSI
### Problem Description

The lastest integrations https://github.com/facebook/react-native/compare/7cece3423...189c2c895 added the ability to access properties with symbol keys through JSI https://github.com/facebook/react-native/commit/9010bfe457b77862024214ce6210504ff1786ef5. This broke us and we need to add these functions to the windows-side to get rid of the overrides (may also need to add the function to hermes-windows)

Errors:

`C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\memory(3415,0): Error C2259: Microsoft::JSI::anonymous-namespace'::NapiJsiRuntime': cannot instantiate abstract class`

`C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\memory(3416,35): Error C2259: 'Microsoft::JSI::SystemChakraRuntime': cannot instantiate abstract class`

	Assertion failed!
	
	Program: ...round\windows\x64\Debug\playground\AppX\hermes.dll
	File: D:\a\1\s\include\hermes/VM/HermesValue.h
	Line: 409
	
	Expression: isSymbol()
	
	For information on how your program can cause an assertion
	failure, see the Visual C++ documentation on asserts
	
	(Press Retry to debug the application - JIT must be enabled)playground.exe has triggered a breakpoint.
	
	Assertion failed!
	
	Program: ...round\windows\x64\Debug\playground\AppX\hermes.dll
	File: D:\a\1\s\include\hermes/VM/SymbolID.h
	Line: 116
	
	Expression: id <= LAST_INVALID_ID && "ID outside valid range"
	
	For information on how your program can cause an assertion
	failure, see the Visual C++ documentation on asserts
	
	(Press Retry to debug the application - JIT must be enabled)playground.exe has triggered a breakpoint.
	
	Assertion failed!
	
	Program: ...round\windows\x64\Debug\playground\AppX\hermes.dll
	File: D:\a\1\s\lib\VM\DictPropertyMap.cpp
	Line: 97
	
	Expression: symbolID.isValid() && "looking for an invalid SymbolID"
	
	For information on how your program can cause an assertion
	failure, see the Visual C++ documentation on asserts

### Steps To Reproduce

1. Remove override files (JSCRuntime.cpp, Jsi.h, decorator.h )
2. Run the playground app

### Expected Results

Playground to run w/o issues

### CLI version

npx react-native --version

### Environment

```markdown
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
    Memory: 49.85 GB / 63.85 GB
  Binaries:
    Node: 14.17.5 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.2.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.17763.0, 10.0.18362.0, 10.0.19041.0, 10.0.22000.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.11.32126.315 (Visual Studio Enterprise 2019)
  Languages:
    Java: 1.8.0_302 - C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

10.0.19041

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2019

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

_No response_

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
Integrations from react-native added createPropNameIDFromSymbol to JSI. Now that V8/hermes-windows have taken the changes from upstream, we should be able to remove forking in our repository by bumping Hermes version to 0.11.0-ms.6.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
Resolves #9703 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9940)

### CODE REVIEW

File: vnext/Shared/InspectorPackagerConnection.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ class RemoteConnection final : public facebook::react::IRemoteConnection {
 class RemoteConnection2 final : public facebook::react::IRemoteConnection2 {
  public:
   RemoteConnection2(int64_t pageId, const InspectorPackagerConnection &packagerConnection);
-  void onMessage(std::unique_ptr<facebook::react::IHermesString> message) override;
+  void onMessage(std::string message) override;
```
Comment (TatianaKapos) [inline vnext/Shared/InspectorPackagerConnection.h:64]: Looks like IHermesString was removed via https://github.com/microsoft/hermes-windows/pull/89

**Comment by tudorms** [comment]
To double check, is the plan still to backport this to 0.69?

**Review by tudorms — APPROVED** [review]
:shipit:

**Comment by TatianaKapos** [comment]
> To double check, is the plan still to backport this to 0.69
yes! Will make the PR today to match versions unless it's not needed.

**Pull request merged**